pkg/component/runtime/service.go

```diff
@@ -364,14 +363,14 @@ func (s *serviceRuntime) awaitCheckin(ctx context.Context, comm Communicator, ti
 	}
 }
 
-func (s *serviceRuntime) processNewComp(newComp component.Component, comm Communicator) {
+func (s *serviceRuntime) processNewComp(newComp component.Component, comm Communicator, sendObs bool) {
```

Comment:
Could you add a comment here explaining when it makes sense to pass `sendObs` as `true` and when to pass it as `false`?

Reply:
The user of the additional flag here is a bit ugly. I'll split this function in two specific for each use and will add the comment.